### PR TITLE
Unhide unknown spans

### PR DESCRIPTION
**Pull request by oli-obk**
r? @estebank

### CODE REVIEW

**Review by davidtwco — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Failed in https://github.com/rust-lang/rust/pull/104123.

@bors r-

File: src/test/ui/associated-types/associated-types-eq-hr.stderr [hunk 1]
```diff
@@ -14,9 +14,6 @@ LL |     type A = &'a usize;
 note: required by a bound in `foo`
   --> $DIR/associated-types-eq-hr.rs:45:36
    |
-LL | fn foo<T>()
-   |    --- required by a bound in this
-LL | where
```
Comment (estebank) [inline src/test/ui/associated-types/associated-types-eq-hr.stderr:19]: Why are the changes above causing these?

File: compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs [hunk 2]
```diff
@@ -2438,17 +2438,6 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
...
-                    let sm = tcx.sess.source_map();
-                    let same_line =
-                        match (sm.lookup_line(ident.span.hi()), sm.lookup_line(span.lo())) {
-                            (Ok(l), Ok(r)) => l.line == r.line,
-                            _ => true,
-                        };
-                    if !ident.span.overlaps(span) && !same_line {
-                        multispan.push_span_label(ident.span, "required by a bound in this");
-                    }
-                }
```
Comment (oli-obk) [inline compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs:2451]: @estebank I removed this code piece, (see 402714f4a7efa28ea12fa0db1864a0ca81990be4), causing all the removals. The information is redundant as there's always a follow up span that actually points to the bound.

Comment (estebank) [reply compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs:2451]: The bound is always pointed, yes, but that span was in order to let you scan the cli output and understand what item the bound was coming from, regardless of whether the bound is defined in the same line of as the ident or not. It feels like a regression to me, but maybe people hate it?

**Comment by oli-obk** [comment]
I re-added the label, but avoid emitting it in case of dummy spans.

**Comment by estebank** [comment]
> I re-added the label, but avoid emitting it in case of dummy spans.

Makes sense! *Now* I'm intrigued as to why some of those where DUMMY_SP to begin with, nothing I saw in the previous tests gave me the impression they'd be, but I'm ok with proactive checking. The whole diagnostics machinery should treat all spans as `Option<Span>` and give enough tools to decide what to show, instead of what we're doing now :-/

@bors r+

**Comment by oli-obk** [comment]
Yea... the idents should have spans, but I didn't quickly find anything so I opted for this instead.

**Pull request merged**